File: src/main/java/com/ptah/repository/userprofiling/UserRepository.java
package com.ptah.repository.userprofiling;

import com.ptah.entity.userprofiling.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByLogin(String login);
}

File: backend/src/main/kotlin/com/github/jarryzhou/ptah/repository/userprofiling/UserRepository.kt
package com.github.jarryzhou.ptah.repository.userprofiling

import com.github.jarryzhou.ptah.entity.userprofiling.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User?, Long?> {
    fun findByLogin(login: String?): User?
}
